### PR TITLE
Deprecate offix into graphback-datasync package

### PR DESCRIPTION
**Pull request by ssd71**
Work is in progress

### CODE REVIEW

**Comment by wtrocki** [comment]
@ssd71 Let's avoid huge PR's like previous one. Ideally we should have max 1-2 days PR's. If we split work to only move and rename this PR could work nicely. 

THere is no documentation for offix and this is not going to affect anything

File: packages/graphback-datasync/src/providers/DatasyncKnexDataProvider.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { GraphQLObjectType } from 'graphql';
```
Comment (wtrocki) [inline packages/graphback-datasync/src/providers/DatasyncKnexDataProvider.ts:1]: I will keep it mongo only for the moment. 
We will focus on Mongo and bring knex at later point

File: packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import { getDatabaseArguments } from '@graphback/core';
+import { ObjectId } from 'mongodb';
+import { MongoDBDataProvider } from '@graphback/runtime-mongo';
+
+/**
+ * Mongo provider that contains special handlers for offix conflict resolution format:
+ *
+ * https://offix.dev/docs/conflict-server#structure-of-the-conflict-error
+ */
+export class DatasyncMongoDBDataProvider<Type = any, GraphbackContext = any> extends MongoDBDataProvider<Type, GraphbackContext> {
```
Comment (wtrocki) [inline packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts:12]: DataSync

**Comment by wtrocki** [comment]
@ssd71 Let's avoid huge PR's like previous one. Ideally we should have max 1-2 days PR's. If we split work to only move and rename this PR could work nicely. 

THere is no documentation for offix and this is not going to affect anything

File: packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts (hunk 2 above)
Comment (ssd71) [reply packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts:12]: I am a bit puzzled as to what to do here, do you mean to change docstring or name of class

**Comment by ssd71** [comment]
> @ssd71 Let's avoid huge PR's like previous one. Ideally we should have max 1-2 days PR's. If we split work to only move and rename this PR could work nicely.
> 
> THere is no documentation for offix and this is not going to affect anything

Agreed

File: packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts (hunk 2 above)
Comment (wtrocki) [reply packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts:12]: I ment rename. You can rename to camelcase. We can do it afterwards.

**Pull request merged**